plugins/prometheus_plugin/prometheus_plugin.cpp

```diff
@@ -87,11 +119,28 @@ namespace eosio {
             add_plugin_metric(metrics->dropped_trxs);
          }
 
+         void add_plugin_metrics(std::shared_ptr<producer_plugin_metrics> metrics) {
+            add_plugin_metric(metrics->unapplied_transactions);
+            add_plugin_metric(metrics->blacklisted_transactions);
+            add_plugin_metric(metrics->blocks_produced);
+            add_plugin_metric(metrics->trxs_produced);
+            add_plugin_metric(metrics->last_irreversible);
+            add_plugin_metric(metrics->block_num);
+            add_plugin_metric(metrics->subjective_bill_account_size);
+            add_plugin_metric(metrics->subjective_bill_block_size);
+            add_plugin_metric(metrics->scheduled_trxs);
```

Comment:
Right now you have to modify both `prometheus_plugin` and `producer_plugin` every time there is a new metric. Instead seems like you could do something like:

```
struct producer_plugin_metrics {
   runtime_metric unapplied_transactions{metric_type::gauge, "unapplied_transactions", "unapplied_transactions", 0};
   runtime_metric blacklisted_transactions{metric_type::gauge, "blacklisted_transactions", "blacklisted_transactions", 0};


   std::vector<std::reference_wrapper<runtime_metric>> metrics;

   producer_plugin_metrics() { 
      metrics.emplace_back(std::ref(unapplied_transactions)); 
      // etc.
   }
};
```

Then you could just loop in the `prometheus_plugin`.